pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -612,7 +611,7 @@ func AdsPushAll(s *DiscoveryServer) {
 // Primary code path is from v1 discoveryService.clearCache(), which is added as a handler
 // to the model ConfigStorageCache and Controller.
 func (s *DiscoveryServer) AdsPushAll(version string, req *model.PushRequest) {
-	if !req.Full {
+	if !req.Full || req.NamespaceUpdatedByPeerAuthn != nil {
```

Comment:
revert it.

When req.Full = true it should always do full push, and if `NamespaceUpdatedByPeerAuthn ` != nil, req.Full should be always set to true.

Reply:
done..